controller/grpc.go

```diff
@@ -251,12 +253,16 @@ func (ss *ScanService) scannerRegister(data *share.ScannerRegisterData) error {
 			CVEDBCreateTime: data.CVEDBCreateTime,
 			CVEDBEntries:    len(data.CVEDB),
 		}
-		clusHelper.PutScannerTxn(txn, &dbVerScanner)
+		if err := clusHelper.PutScannerTxn(txn, &dbVerScanner); err != nil {
```

Comment:
To handle the returned error here, in kv/helper.go,
```
func (m clusterHelper) PutScannerTxn(txn *cluster.ClusterTransact, s *share.CLUSScanner) error {
	key := share.CLUSScannerKey(s.ID)
	value, _ := enc.Marshal(s)
	txn.Put(key, value)
	return nil
}
```
should be change to something like
```
func (m clusterHelper) PutScannerTxn(txn *cluster.ClusterTransact, s *share.CLUSScanner) error {
	key := share.CLUSScannerKey(s.ID)
	value, err := enc.Marshal(s)
        if err != nil {
                return err
        }
	txn.Put(key, value)
	return nil
}
```
so that it doesn't always return nil, correct?

Reply:
fixed